api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.gcp_authn.v3;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/config/core/v3/http_uri.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.gcp_authn.v3";
+option java_outer_classname = "GcpAuthnProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/gcp_authn/v3;gcp_authnv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: GCP authentication]
+// GCP authentication :ref:`configuration overview <config_http_filters_gcp_authn>`.
+// [#extension: envoy.filters.http.gcp_authn]
+
+// Configuration.
+message GcpAuthnFilterConfig {
+  // The HTTP URI to fetch tokens from GCE Metadata Server(https://cloud.google.com/compute/docs/metadata/overview).
+  config.core.v3.HttpUri http_uri = 1;
+
+  // Retry policy for fetching tokens.
+  config.core.v3.RetryPolicy retry_policy = 2;
```

Comment:
are these required, if not what will be the default?

Reply:
Thanks for the review! This is a good question.
We plan to start with no retry for MVP but do want to keep it and find proper default setting as we see fit in production.

Reply:
This comment applies to both fields, you should mark `http_uri` as required though?

Reply:
I added the validation rule to force `http_uri` as required.
`optional` field is not supported in Envoy yet: code generator protoc-gen-api_proto_plugin hasn't been updated to support optional fields in proto3